admin/test/test_installer.py

```diff
@@ -0,0 +1,103 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+"""
+Tests for ``admin.installer``.
+"""
+from textwrap import dedent
+
+from pyrsistent import PClass, field, pmap_field
+
+from flocker.testtools import (
+    TestCase
+)
+
+from ..installer._images import (
+    _packer_amis, _PackerOutputParser,
+)
+
+
+class ParserData(PClass):
+    input = field(type=unicode, mandatory=True)
+    output = pmap_field(key_type=unicode, value_type=unicode, optional=False)
+
+
+PACKER_OUTPUT_US_WEST_1 = ParserData(
+    input=u"""\
+1450420216,,ui,say,Build 'amazon-ebs' finished.
+1450420216,,ui,say,\n==> Builds finished. \
+The artifacts of successful builds are:
+1450420216,amazon-ebs,artifact-count,1
+1450420216,amazon-ebs,artifact,0,builder-id,mitchellh.amazonebs
+1450420216,amazon-ebs,artifact,0,id,us-west-1:ami-e098f380
+1450420216,amazon-ebs,artifact,0,string,AMIs were created:\n\n\
+us-west-1: ami-e098f380
+1450420216,amazon-ebs,artifact,0,files-count,0
+1450420216,amazon-ebs,artifact,0,end
+1450420216,,ui,say,--> amazon-ebs: AMIs were created:\n\n\
+us-west-1: ami-e098f380
+""",
+    output={u"us-west-1": u"ami-e098f380"}
+)
+
+PACKER_OUTPUT_US_ALL = ParserData(
+    input=u"""\
+1450420216,,ui,say,Build 'amazon-ebs' finished.
+1450420216,,ui,say,\n==> Builds finished. \
+The artifacts of successful builds are:
+1450420216,amazon-ebs,artifact-count,1
+1450420216,amazon-ebs,artifact,0,builder-id,mitchellh.amazonebs
+1450420216,amazon-ebs,artifact,0,id,\
+us-east-1:ami-dc4410b6%!(PACKER_COMMA)\
+us-west-1:ami-e098f380%!(PACKER_COMMA)\
+us-west-2:ami-8c8f90ed
+1450420216,amazon-ebs,artifact,0,string,AMIs were created:\n\n\
+us-east-1: ami-dc4410b6\n\
```

Comment:
`\n\` is a really strange pattern. "Add a newline, and then ignore the next newline" perhaps just remove `\n\` if it occurs at the end of a line?

```
>>> c = u"""
... cat\n\
... """
>>> d = u"""
... cat
... """
>>> c == d
True
>>> c
u'\ncat\n'
>>> d
u'\ncat\n'
>>>
```